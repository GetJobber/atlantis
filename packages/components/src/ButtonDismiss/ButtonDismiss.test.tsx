import React from "react";
import renderer from "react-test-renderer";
import { cleanup, fireEvent, render } from "@testing-library/react";
import { ButtonDismiss } from ".";

afterEach(cleanup);

it("renders a ButtonDismiss", () => {
  const tree = renderer
    .create(<ButtonDismiss onClick={undefined} ariaLabel="Close" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("it should call the handler with the new value", () => {
  const clickHandler = jest.fn();
  const { getByRole } = render(
    <ButtonDismiss onClick={clickHandler} ariaLabel="Close" />,
  );

  fireEvent.click(getByRole("button"));
  expect(clickHandler).toHaveBeenCalled();
});
