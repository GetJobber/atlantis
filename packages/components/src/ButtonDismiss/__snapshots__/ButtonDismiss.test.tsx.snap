// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a ButtonDismiss 1`] = `
<div>
  <button
    aria-label="Close"
    class="button base subtle tertiary buttonChildren"
    type="button"
  >
    <svg
      data-testid="remove"
      style="fill: var(--color-icon); display: inline-block; vertical-align: middle; width: 24px; height: 24px;"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 10.586 6.707 5.293a1 1 0 0 0-1.414 1.414L10.586 12l-5.293 5.293a1 1 0 0 0 1.414 1.414L12 13.414l5.293 5.293a1 1 0 0 0 1.414-1.414L13.414 12l5.293-5.293a1 1 0 1 0-1.414-1.414L12 10.586Z"
      />
    </svg>
  </button>
</div>
`;
